mne/io/curry/curry.py

```diff
@@ -0,0 +1,281 @@
+    # The first 3 column seem to contain the event information
+    curry_events = np.array(events_dict["NUMBER_LIST"], dtype=int)[:, 0:3]
+
+    if event_ids is not None:
+        idx = np.array([i in event_ids for i in curry_events[:, -1]])
+        curry_events = curry_events[idx]
+
+    return curry_events
+
+
+def read_raw_curry(input_fname, preload=False):
+    """
+    Read raw data from Curry files.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to a curry file with extensions .dat, .dap, .rs3, .cdt, cdt.dpa,
+        .cdt.cef or .cef.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory). If the curry file
+        is stored in ASCII data format, then preload will automatically be
+        set to True.
+
+    Returns
+    -------
+    raw : instance of RawCurry
+        A Raw object containing Curry data.
+
+    """
+    # we don't use os.path.splitext to also handle extensions like .cdt.dpa
+    fname_base, ext = input_fname.split(".", maxsplit=1)
+
+    curry_vers = _get_curry_version(ext)
+    _check_missing_files(fname_base, curry_vers)
+
+    info, n_trials, n_samples, curry_vers, ascii = _read_curry_info(fname_base,
+                                                                    curry_vers)
+
+    raw = RawCurry(fname_base + DATA_FILE_EXTENSION[curry_vers], info,
+                   n_samples, ascii, preload)
+
+    return raw
+
+
+class RawCurry(BaseRaw):
+    """
+    Raw object from Curry file.
+
+    Parameters
+    ----------
+    data_fname : str
+        Path to the Curry data file (.dat or .cdt).
+    info : instance of mne.Info
+        An instance of mne.Info as returned by mne.io.curry._read_curry_info.
+    n_samples : int
+        The number of samples of the curry data file.
+    ascii : bool (default False)
+        If the loaded data file is coded in ASCII, ascii must be set to True.
+        Else must be False.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory) If ascii is True,
+        preload will automatically be set to True.
+    %(verbose)s
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+
+    """
+
+    def __init__(self, data_fname, info, n_samples, ascii,
+                 preload=False, verbose=None):
```

Comment:
the signature of the constructor should match the one from read_raw_curry

it can feel weird but it is consistent with all readers.

read_raw_curry(*args, **kwargs)
RawCurry(*args, **kwargs)

should be equivalent

Reply:
So, if that's the case, the most efficient way would be to only keep this in read_raw_curry:
```python
def_read_raw_curry(input_fname, preload=False, verbose=None):

    raw = RawCurry(input_fname, preload=False, verbose=None)

    return raw
```
And perform all other steps in `RawCurry.__init__`...

Should i do it like that?